File: typescipt/existing/src/language/expressions/functions/dayFunction.ts



namespace Lexy.Compiler.Language.Expressions.Functions;

public class DayFunction : SingleArgumentFunction
{
   public const string Name = "DAY";

   protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   private DayFunction(Expression valueExpression, SourceReference reference)
     : base(valueExpression, reference)
   {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression expression)
   {
     return new DayFunction(expression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/DayFunction.cs
using Lexy.Compiler.Language.VariableTypes;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class DayFunction : SingleArgumentFunction
{
    public const string Name = "DAY";

    protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

    private DayFunction(Expression valueExpression, ExpressionSource source)
        : base(Name, valueExpression, source, PrimitiveType.Date, PrimitiveType.Number)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression expression)
    {
        return new DayFunction(expression, source);
    }
}